test/parallel/test-fs-readfile-fd-offset.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+
+const fileLength = fs.statSync(__filename).size;
+
+['r', 'a+'].forEach(mode => {
+  const fd = fs.openSync(__filename, mode);
```

Comment:
I would use a file in `common.tmpDir` for a+ because `__filename` is not going to work when the file is owned by a different user.